### DISCUSSION TITLE
When to adapt to miyoo mini plus

### DISCUSSION THREAD

**Discussion by cjv123** [OnionUI/Onion#704] [Q&A] [answered]
When to adapt to miyoo mini plus

**Comment by Aemiii91** [comment]
**4.1** is expected to release in a few days and **will not** be compatible with Mini+.  

**4.2** is expected to hit beta before end-of-month, and will be the "Mini+" update, being compatible with both devices.

**Reply by ElusiveZatchmo** [reply]
I just received my Miyoo Mini+ and I'm excited to install Onion on it when it becomes available for it.

**Comment by TerryZ** [comment]
Can't wait to install Onion to my Mini+

**Comment by raeldor** [comment]
Will we just be able to swap the SD card from Miyoo Mini to Mini Plus and back again after 4.2?

**Reply by marosige** [reply]
I don't have the mini, just the mini+ but it seems like it will be swappable!

**Comment by zapdostimetheory** [comment]
I would love to help contribute in anyway I can. I have the Miyoo Mini Plus and am willing to help with testing. I am an amateur programmer but can take on tedious tasks.

**Comment by ghost** [comment]
I just got my Miyoo Mini+ and would love to help beta test 4.2, im a full stack dev and happy to help if i can

**Comment by marosige** [comment]
I'm using the V4.2-beta on my Mini+ for a day now and I love it! Really nice work Onion team! Wifi is not working but the volume buttons (with brightness control) are working nice. Haven't found any issues while using it.
I made a video about running it: https://youtu.be/k-ZWCYtPRxc

To get Onion OS V4.2-beta on the Miyoo Mini+ you have to build your own OS version.
It's not hard at all, all you need is a little knowledge in software development.
I did it on Mac OS, it should be the same in linux.
If you use windows I recommend to use a VM with linux. More info here: https://github.com/OnionUI/Onion/wiki/Setup#environment

Download Docker if you don't have it: https://www.docker.com/products/docker-desktop/
Build your own V4.1.3 version: https://github.com/OnionUI/Onion/wiki/Setup#building

To build your own V4.2-beta version all you have to change is to check out the V4.2-beta branch with `git checkout -b dev-V4.2-beta`

The whole process with the checkout in terminal:
```
git clone https://github.com/OnionUI/Onion.git
cd Onion/
git checkout -b dev-V4.2-beta
make git-submodules
make with-toolchain
```

To install it just follow the instructions here: https://github.com/OnionUI/Onion/wiki/Installing-a-build

**Reply by zapdostimetheory** [reply]
I’m going to try this when I get home; never used Docker before so this is the best way to start lol.

**Reply by marosige** [reply]
You only have to install docker and the makefile takes care of everything. If you are using windows and go with the VM route, the wiki links a linux image with docker pre installed so it should be easy.

**Reply by Ttch21** [reply]
Can I recommend just editing the docker command in the makefile to have `--platform linux/amd64` rather than wasting time and space setting up a linux VM

**Reply by Zoolander157** [reply]
First off gotta say thanks for your original post.  It got me pointed in the right direction.  I also saw you edited your original post and wanted to be sure it'll work for as many people as possible.  The process that worked for me is below.

`git clone -b dev-V4.2-beta https://github.com/OnionUI/Onion.gitt`
`cd Onion/`
`git checkout dev-V4.2-beta`
`make git-submodules`
`make with-toolchain`

And I'm not 100% the "git checkout" is necessary either, but I don't think it is hurting anything.  The -b for the checkout will throw an error.

**Reply by zapdostimetheory** [reply]
Took a little bit of time for me. The joys of learning WSL.. then having WSL 2 woes with Debian (it wouldn't switch into 2 lol).. I did the Kali distro instead and that worked. I got the beta branch and was able to install it to my system. Thank you for the help all! Testing it out now.

**Comment by raeldor** [comment]
I tried following these build instructions on the Mac, but the version still said 4.14 after it was done.  When I tried using it in my Mini+ the games were crashing and volume was not working.  Did I miss something?  Thanks.

**Reply by Zoolander157** [reply]
Try cloning from the dev-V4.2-beta repo using 
`git clone -b dev-V4.2-beta https://github.com/OnionUI/Onion.git`
Then follow the steps same as listed.

**Reply by raeldor** [reply]
That worked, thank you.  I guess doing -b there is different from doing git checkout?

**Reply by Zoolander157** [reply]
Sweet glad that worked for you. The -b tells it to get just the dev 4.2 beta branch only. I think the standard way is only pulling from the main branch dev build. At least for me that's what it did.

**Reply by raeldor** [reply]
Thank you.  Being able to swap my SD card between the two and have my progress carry is invaluable.  As is the short hold on the power button to save state and power off during a game.  I love Onion. :)

**Comment by Aemiii91** [comment, marked as answer]
Official beta is now live: https://github.com/OnionUI/Onion/releases/tag/dev-v4.2.0-beta

**Reply by turbonetlink** [reply]
quando sai a versão estavel ?
